statsmodels/stats/tests/test_outliers_influence.py

```diff
@@ -14,3 +16,11 @@ def test_reset_stata():
     stat = reset_ramsey(res, degree=4)
     assert_almost_equal(stat.fvalue[0, 0], 1.52, decimal=2)
     assert_almost_equal(stat.pvalue, 0.2221, decimal=4)
+
+
+def test_variance_inflation_factor():
+    # Test Case: perfect collinear with constant component
+    exog = np.stack([np.arange(8), 10. - np.arange(8)]).T
+    exog = add_constant(exog)
+    vif = variance_inflation_factor(exog, 0)
+    assert_almost_equal(vif, 0., decimal=4)
```

Comment:
Is this different what one would get if has_constant=1 (or True)?

Reply:
If so, could you add that case as well.

Reply:
This test does not relate to #5973 directly.
This test considers the case when `hasconstant=True`, which always worked correctly.
Setting `hasconstant=True` explicitly will return the same result.